pkg/endpoint/bpf.go

```diff
@@ -974,51 +974,63 @@ func (e *Endpoint) InitMap() error {
 	return policymap.Create(e.policyMapPath())
 }
 
-// deleteMaps releases references to all BPF maps associated with this
-// endpoint.
-//
-// For each error that occurs while releasing these references, an error is
-// added to the resulting error slice which is returned.
+// deleteMaps deletes the endpoint's entry from the global
+// cilium_(egress)call_policy maps and removes endpoint-specific cilium_calls_,
+// cilium_policy_ and cilium_ct{4,6}_ map pins.
 //
-// Returns nil on success.
+// Call this after the endpoint's tc hook has been detached.
```

Comment:
Nothing prevents this code being reordered in the future via some refactorings. Have you considered defining a simple tear down state transition to define the expected flow?

Reply:
> Nothing prevents this code being reordered in the future via some refactorings

The whole bringup and teardown flow is slated for refactoring (by us), but as with any refactoring, ordering matters and needs to be respected. The comments I sprinkled should at least allude to this fact, both to the person performing the refactor as well as the reviewer.

This code is in no state to have unit/integration tests added (but many of the building blocks already do!), so any refactoring effort should at least try to improve on this while obviously keeping the ordering the same. I didn't mark it as such, but this PR is to be backported to 1.15 after it gained some more time in the tree, so I had to hold off on the more invasive changes. 

For example, most of the stuff in `loader.Unload()` is really Endpoint/bpf_lxc-specific and should be moved to package endpoint, but we haven't mapped out the bigger picture yet.

> Have you considered defining a simple tear down state transition to define the expected flow?

What would this look like? What kind of state transition? This sounds really hand-wavey.

Reply:
Maybe we can just add a `HERE BE DRAGONS` comment?

Reply:
> Maybe we can just add a `HERE BE DRAGONS` comment?

Looking at the code, this feels rather obvious, so I'm not sure if a comment like that would add much.